xarray/coding/variables.py

```diff
@@ -348,20 +375,51 @@ def _scale_offset_decoding(data, scale_factor, add_offset, dtype: np.typing.DTyp
     return data
 
 
-def _choose_float_dtype(dtype: np.dtype, has_offset: bool) -> type[np.floating[Any]]:
+def _choose_float_dtype(
```

Comment:
This function checks for the most appropriate dtype to use when encoding/decoding in `CFScaleOffsetCoder`.